Library/Homebrew/rubocops/lines.rb

```diff
@@ -102,6 +102,33 @@ class OptionDeclarations < FormulaCop
         def audit_formula(_node, _class_node, _parent_class_node, body_node)
           problem "Use new-style option definitions" if find_method_def(body_node, :options)
 
+          if formula_tap == "homebrew-core"
+            # Use of build.with? implies options, which are forbidden in homebrew/core
+            find_instance_method_call(body_node, :build, :without?) do
+              problem "Formulae in homebrew/core should not use `build.without?`."
+            end
+            find_instance_method_call(body_node, :build, :with?) do
+              problem "Formulae in homebrew/core should not use `build.with?`."
+            end
+
+            return
+          end
+
+          # Matches `depends_on "foo" if build.with?("bar")` or depends_on "foo" if build.without?("bar")`
+          depends_on_build_regex = /depends_on .+ (if build\.with(out)?\?\(["']\w+["']\))/
+
+          find_instance_method_call(body_node, :build, :with?) do |n|
+            next unless match = n.parent.source.match(depends_on_build_regex)
```

Comment:
Not sure I understand this but; isn't it just done by `find_instance_method_call`? What is the extra match doing?

Reply:
`find_instance_method_call` gets all occurrences of `build.with?`. This includes any instances within the `install` block. The match then filters out only the instances that are part of a `depends_on` line, as that's what the cop is checking.

I wanted to have a non-regex way to do this but got stuck. One option that I thought would work would be to use `def_node_search` or `def_node_match` but I couldn't figure out how to use them properly (and didn't find much online).

Now that I'm looking at it it's possible I could, instead of using a regex, check that `n.parent`'s type is `if` and that it's parent's method name is `depends_on`. I'll look into it later today.

Reply:
> I wanted to have a non-regex way to do this but got stuck.

Good enough for me. Could you add a `# TODO` for future refactoring if you decide to merge as-is?